onnxruntime/test/mlas/unittest/test_dynamic_qgemm.cpp

```diff
@@ -96,45 +142,73 @@ class MlasDynamicQgemmTest {
       }
     }
 
+    std::fill(C, C + M * N * BatchSize, 0.0f);
+    MlasDynamicQGemmBatch(shape, params.data(), BatchSize, threadpool);
+
     // Validate results
-    for (size_t i = 0; i < M * N * BatchSize; ++i) {
-      float abs_c_ref = std::abs(CRef[i]);
-      float dynamic_rel_tol = (K <= 4) ? 0.05f : 0.03f;
-      float rel_tol = dynamic_rel_tol * std::max(abs_c_ref, 1.0f);
-      float abs_tol = 3.0f;
-      float allowed = std::max(rel_tol, abs_tol);
-      float diff = std::abs(C[i] - CRef[i]);
-      ASSERT_LE(diff, allowed);
-    }
+    auto validate = [&](const char* tag) {
+      SCOPED_TRACE(tag);
+      for (size_t i = 0; i < M * N * BatchSize; ++i) {
+        float abs_c_ref = std::abs(CRef[i]);
+        float dynamic_rel_tol = (K <= 4) ? 0.05f : 0.03f;
+        float rel_tol = dynamic_rel_tol * std::max(abs_c_ref, 1.0f);
+        float abs_tol = 3.0f;
+        float allowed = std::max(rel_tol, abs_tol);
+        float diff = std::abs(C[i] - CRef[i]);
+        ASSERT_LE(diff, allowed);
+      }
+    };
+
+    validate(run_tag);
+  }
+};
+
+class MlasDynamicQgemmSingleThreadTest : public MlasDynamicQgemmTestBase {
+ public:
+  void Test(size_t M, size_t N, size_t K, size_t BatchSize) {
+    // Currently, MlasDynamicQGemmBatch() and associated functions require SME or else they are no-ops.
+    if (!MLAS_CPUIDINFO::GetCPUIDInfo().HasArm_SME())
```

Comment:
Could we use the MLAS APi to check if dynamic Q Gemm functionality is available ?